File: postgraduate/src/main/webapp/js/controller/TrainController.js
app.controller('TrainController',function($scope,$controller,TrainService){
	//增加保存
	$scope.save=function(entity){
		var object=null;		
		object=TrainService.insert(entity);
		object.success(
				function(response){
					if(response.success){
						alert(response.message);
					}else{
						alert(response.message);
					}
				}
		);
	}   
	//单条查询
	$scope.selectOne=function(id){
		TrainService.selectOne(id).success(
			function(response){
				$scope.entity=response;
			}		
		);
	}
	$scope.reloadList=function(){
		 //切换页码  
		$scope.search( $scope.paginationConf.currentPage, $scope.paginationConf.itemsPerPage);
	}
	
	//分页控件配置 
	$scope.paginationConf = {
		 currentPage: 1,
		 totalItems: 1,
		 itemsPerPage: 1,
		 perPageOptions: [1, 2, 3, 4, 5],
		 onChange: function(){
		        	 $scope.reloadList();//重新加载
		 }
	};

   //分页
	$scope.findPage=function(page,rows){	
		TrainService.findPage(page,rows).success(
				function(response){
					$scope.list=response.rows;	
					$scope.paginationConf.totalItems=response.total;//更新总记录数
				}			
		);
	}
	
	//审核
	$scope.updateState=function(id,state){
		TrainService.updateState(id,state).success(
				function(response){
					if(response.success){
						$scope.reloadList();//刷新列表
					}else{
						alert("失败");
					}	
				}
		);
	}
	
	$scope.searchEntity={};//定义搜索对象
	//条件查询
	$scope.search=function(page,rows){
		TrainService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.paginationConf.totalItems=response.total;//总记录数 
				$scope.list=response.rows;//给列表变量赋值 

			}		
		)
	}
	//批量设置课程
	$scope.selIds=Ids;
	$scope.insertLesTra=function(ids,traid){
		$scope.selIds=Ids;
		TrainService.insertLesTra($scope.selIds,traid).success(
			function(response){
				if(response.success){
					$scope.reloadList();
				}
			}	
		);
	}
	$scope.state=['未审核','已审核','审核未通过'];//商品状态
});

File: postgraduate/src/main/java/com/postgraduate/controller/TrainController.java
package com.postgraduate.controller;


import java.util.List;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.postgraduate.pojo.TbLesTra;
import com.postgraduate.pojo.TbStudent;
import com.postgraduate.pojo.TbTrain;
import com.postgraduate.service.TrainService;

import entity.PageResult;
import entity.Result;

@RestController
@RequestMapping("/Train")
public class TrainController {
	@Autowired
	private TrainService trainService;
	
	/**
	 * 增加培养方案
	 * @return
	 */
	@RequestMapping("/insert")
	public Result insert(@RequestBody TbTrain tbTrain) {
		try {
			trainService.insert(tbTrain);
			return new Result(true,"增加成功");
		}catch(Exception e) {
			e.printStackTrace();
			return new Result(false,"增加失败");
		}
	}
	/**
	 * 课程设置
	 * @return
	 */
	@RequestMapping("/insertLesTra")
	public Result insertLesTra(int[] lesid,int traid) {
		try {
			for(int les:lesid) {
				trainService.insertLesTra(les,traid);
			}
			return new Result(true,"增加成功");
		}catch(Exception e) {
			e.printStackTrace();
			return new Result(false,"增加失败");
		}
	}
	/**
	 * 查询所有培养方案
	 * @return
	 */
	@RequestMapping("selectAll")
	public List<TbTrain> selectAll(){
		return trainService.selectAll();
	}
	/**
	 * 查询单个培养方案
	 * 
	 */
	@RequestMapping("/selectOne")
	public TbTrain selectOne(int id) {
		return trainService.selectOne(id);
	}
	/**
	 * 返回分页
	 * 
	 */
	@RequestMapping("/findPage")
	public PageResult findPage(int page,int rows) {
		return trainService.findPage(page, rows);
	}
	/**
	 * 审核培养方案
	 * 
	 */

	@RequestMapping("/updateState")
	public Result updateState(int id,int state) {
		try {
			TbTrain tbtrain = new TbTrain();
			tbtrain.setId(id);
			tbtrain.setState(state);
			trainService.updateState(tbtrain);
			return new Result(true,"修改成功");
		}catch(Exception e){
			e.printStackTrace();
			return new Result(false,"修改失败");
		}
	}
	/**
	 * 模糊查询返回分页
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbTrain tbTrain,int page,int rows) {
		return trainService.findPage(tbTrain,page,rows);
	}
}
